### PR TITLE
Clarify Custom Domain directions

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal. This repo contains the content for https://internetcomputer.org and the ICP Developer Documentation, https://internetcomputer.org/docs/. 

If you are submitting a Pull Request for adding or changing content on the ICP Developer Documentation, please make sure that your contribution meets the following requirements:

- [ ] Follows the [developer docs style guide](style-guide.md).
- [ ] Follows the [best practices and guidelines](README.md#best-practices).
- [ ] New documentation pages include [document tags](README.md#document-tags).
- [ ] New documentation pages include [SEO keywords](README#seo-keywords).
- [ ] New documents are in the `.mdx` file format to support the previous two components. 
- [ ] New documents are registered in [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js), otherwise, it will not appear in the
  side navigation bar.
- [ ] Make sure that the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file is
  filled with new documents that you added. This way we can ensure that future Pull Requests are reviewed by the right people.

### CODE REVIEW

File: docs/developer-docs/web-apps/custom-domains/using-custom-domains.mdx [hunk 1]
```diff
@@ -115,139 +124,152 @@ If the call was successful, you will get a JSON response that contains the reque
...
 - Check that there are no `TXT` records for the `_acme-challenge`-subdomain (e.g., by using `dig TXT _acme-challenge.CUSTOM_DOMAIN`). If there are `TXT` records, then they are most likely left-over from previous ACME-challenges by your domain provider. Note that these records often do not show up in your domain management dashboard. Try disabling all TLS/SSL-certificate offerings from your domain provider to remove these records.
 - Check the `ic-domains` file by downloading it directly from your canister (e.g., by opening `CANISTER_ID.icp0.io/.well-known/ic-domains` in your browser or using `curl CANISTER_ID.icp0.io/.well-known/ic-domains` in your terminal).
 
+:::info
+You may need to specify a `host` in your frontend code when you are using a custom domain, as the `HttpAgent` may not be able to automatically infer the host like it can on `icp0.io` and `ic0.app`. To configure your agent, it will look something like this:
+
+```ts
+// Point to icp-api for the mainnet. Leaving host undefined will work for localhost
+const host = isProduction ? 'https://icp-api.io' : undefined;
+const agent = new HttpAgent({ host });
```
[the remark this answers is not in the document]
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/using-custom-domains.mdx:220]: @krpeacock Could you suggest alternate code that uses the new recommended method?

Comment (krpeacock) [inline docs/developer-docs/web-apps/custom-domains/using-custom-domains.mdx:220]:
```suggestion
const agent = await HttpAgent.create({ host });
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 2]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 This guide explains how to configure the DNS records of your domain for three popular
 registrars: [Namecheap](#namecheap), [GoDaddy](#godaddy), and [Amazon Route 53](#amazonroute53).
 
-## Namecheap
+:::caution
+The steps in this guide are prerequisites to using a custom domain. You must complete this DNS configuration before following the [using custom domains guide](using-custom-domains.mdx)
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:16]:
```suggestion
The steps in this guide are prerequisites to using a custom domain. You must complete this DNS configuration before following the ["using custom domains" guide](using-custom-domains.mdx)
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 3]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+:::
 
-In the following, the steps to configure the apex of a domain and
-a subdomain on Namecheap to be used as a custom domain are listed. To illustrate the required
-configuration, the domain `ic-domain.live` and the subdomain `example.ic-domain.live` are used
+:::info
+To illustrate these required configurations, the domain `ic-domain.live` and the subdomain `example.ic-domain.live` are used
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:20]:
```suggestion
To illustrate these required configurations, the apex domain `ic-domain.live` and the subdomain `example.ic-domain.live` are used
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 4]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
-  There are mainly two approaches:
+  There are two options:
 
-  - [_recommended_: Relying on a different DNS provider (e.g., Cloudflare).](#rely-on-an-alternative-dns-provider-recommended)
-  - [Directly configuring the IP addresses of the boundary nodes.](#directly-configure-the-ip-addresses)
+  - [**Recommended**: Relying on a different DNS provider (e.g., Cloudflare).](#rely-on-an-alternative-dns-provider-recommended)
+  - [Directly configuring the IP addresses of the boundary nodes.](#directly-configure-the-ip-addresses)
 
 Directly configuring the IP addresses is simpler compared to relying on another
 DNS provider, but is less resilient and performant. Therefore, it is advised you to use
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:78]:
```suggestion
DNS provider, but is less resilient and performant. Therefore, it is advised to use
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 5]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
-It is explained in this approach using Cloudflare as DNS provider. It works similar
+It is explained in this approach using Cloudflare as a DNS provider. It works similar
 with any other DNS provider that supports `CNAME`, `ALIAS`, or `ANAME` records
 for the apex of a domain.
 
-  - #### Step 1: Create a free account with Cloudflare, click on **Add site** in the top bar of the dashboard.
+  - #### Step 1: Create a free account with Cloudflare; click on **Add site** in the top bar of the dashboard.
 
-  - #### Step 2: Enter your domain (e.g., `ic-domain.online`) and click **Add site**.
+  - #### Step 2: Enter your domain (e.g., `ic-domain.live`) and click **Add site**.
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:89]:
```suggestion
  - #### Step 2: Enter your domain (e.g., `ic-domain.online`) and click **Add site**.
```
Why change the domain name? All the screenshots are for `ic-domain.online` and not `ic-domain.live`.

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 6]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+  succeeded.
 
-  - #### Step 10: On Cloudflare, add the following records to configure your domain:
-    * Create a `CNAME` record for which you set the name field to `@` and the
-    target field to `ic-domain.online.icp1.io`;
-    * Create a `CNAME` record for which you set the host field to `_acme-challenge` and the target field to `_acme-challenge.ic-domain.online.icp2.io`;
-    * Create a `TXT` record for which you set the name field to `_canister-id` and the content field to the canister ID `y5jqt-wqaaa-aaaam-abcoq-cai`.
+  - #### Step 10: On Cloudflare, add the following records to configure your domain:
+    * Create a `CNAME` record for which you set the name field to `@` and the
+    target field to `ic-domain.live.icp1.io`;
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:116]:
```suggestion
    target field to `ic-domain.online.icp1.io`;
```
Same here. I think it is confusing if the screenshot shows a different domain name than the instructions.

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 7]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+    * Create a `CNAME` record for which you set the host field to `_acme-challenge` and the target field to `_acme-challenge.ic-domain.live.icp2.io`;
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:117]:
```suggestion
    * Create a `CNAME` record for which you set the host field to `_acme-challenge` and the target field to `_acme-challenge.ic-domain.online.icp2.io`;
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 8]
```diff
@@ -12,109 +12,116 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+    * Create a `TXT` record for which you set the name field to `_canister-id` and the content field to the canister ID `y5jqt-wqaaa-aaaam-abcoq-cai`.
 
-  The resulting configuration should look similar to the following screenshot:
+  The resulting configuration should look similar to the following screenshot:
 
-  ![DNS Configuration for `ic-domain.online` on Cloudflare](cloudflare-apex.png)
+  ![DNS Configuration for `ic-domain.live` on Cloudflare](cloudflare-apex.png)
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:122]:
```suggestion
  ![DNS Configuration for `ic-domain.online` on Cloudflare](cloudflare-apex.png)
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 9]
```diff
@@ -123,48 +130,46 @@ addresses, you need to remove or update your DNS records. Otherwise, you risk th
...
-    to "@" and the value field to the IPv4-address (e.g., `193.118.63.173`);
-    * Create an `AAAA` record for each IPv6-address by setting the name field
-    to "@" and the value field to the IPv6-address (e.g., `2a0b:21c0:b002:2:5000:59ff:fead:c233`);
-    * Create a `CNAME` record for which you set the name field to `_acme-challenge` and the value field to `_acme-challenge.ic-domain.online.icp2.io`;
-    * Create a `TXT` record for which you set the name field to `_canister-id` and the value field to the canister ID `y5jqt-wqaaa-aaaam-abcoq-cai`.
+    * Create an `A` record for each IPv4-address by setting the name field
+    to "@" and the value field to the IPv4-address (e.g., `193.118.63.173`);
+    * Create an `AAAA` record for each IPv6-address by setting the name field
+    to "@" and the value field to the IPv6-address (e.g., `2a0b:21c0:b002:2:5000:59ff:fead:c233`);
+    * Create a `CNAME` record for which you set the name field to `_acme-challenge` and the value field to `_acme-challenge.ic-domain.live.icp2.io`;
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:145]:
```suggestion
    * Create a `CNAME` record for which you set the name field to `_acme-challenge` and the value field to `_acme-challenge.ic-domain.online.icp2.io`;
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 10]
```diff
@@ -123,48 +130,46 @@ addresses, you need to remove or update your DNS records. Otherwise, you risk th
...
+    * Create a `TXT` record for which you set the name field to `_canister-id` and the value field to the canister ID `y5jqt-wqaaa-aaaam-abcoq-cai`.
 
-    The resulting configuration should look similar to the following screenshot:
+    The resulting configuration should look similar to the following screenshot:
 
-    ![DNS Configuration for `ic-domain.online` on GoDaddy](godaddy-apex-ips.png)
+    ![DNS Configuration for `ic-domain.live` on GoDaddy](godaddy-apex-ips.png)
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:150]:
```suggestion
    ![DNS Configuration for `ic-domain.online` on GoDaddy](godaddy-apex-ips.png)
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 11]
```diff
@@ -123,48 +130,46 @@ addresses, you need to remove or update your DNS records. Otherwise, you risk th
...
 
 ### Subdomain
 To configure a subdomain (e.g., `example.ic-domain.live`), add the following records by clicking on **Add new record**:
-  * Create a `CNAME` record for which you set the host field to `example` and the target field to `example.ic-domain.online.icp1.io`;
-  * Create a `CNAME` record for which you set the host field to `_acme-challenge.example` and the target field to `_acme-challenge.example.ic-domain.online.icp2.io`;
-  * Create a `TXT` record for which you set the host field to `_canister-id.example` and the value field to the canister ID `y5jqt-wqaaa-aaaam-abcoq-cai`.
+  * Create a `CNAME` record for which you set the host field to `example` and the target field to `example.ic-domain.live.icp1.io`;
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:154]:
```suggestion
  * Create a `CNAME` record for which you set the host field to `example` and the target field to `example.ic-domain.online.icp1.io`;
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 5 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:89]: It is confusing to use two different example domains. We should be consistent in the examples used. Can we update the screenshots to all be consistent, or is there a benefit to using two different example domains?

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 12]
```diff
@@ -123,48 +130,46 @@ addresses, you need to remove or update your DNS records. Otherwise, you risk th
...
+  * Create a `CNAME` record for which you set the host field to `_acme-challenge.example` and the target field to `_acme-challenge.example.ic-domain.live.icp2.io`;
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:155]:
```suggestion
  * Create a `CNAME` record for which you set the host field to `_acme-challenge.example` and the target field to `_acme-challenge.example.ic-domain.online.icp2.io`;
```

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx [hunk 13]
```diff
@@ -123,48 +130,46 @@ addresses, you need to remove or update your DNS records. Otherwise, you risk th
...
+  * Create a `TXT` record for which you set the host field to `_canister-id.example` and the value field to the canister ID `y5jqt-wqaaa-aaaam-abcoq-cai`.
 
-  The resulting configuration should look similar to the following screenshot:
+  The resulting configuration should look similar to the following screenshot:
 
-  ![DNS Configuration for `example.ic-domain.online` on GoDaddy](godaddy-subdomain.png)
+  ![DNS Configuration for `example.ic-domain.live` on GoDaddy](godaddy-subdomain.png)
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:160]:
```suggestion
  ![DNS Configuration for `example.ic-domain.online` on GoDaddy](godaddy-subdomain.png)
```

File: docs/developer-docs/web-apps/custom-domains/using-custom-domains.mdx [hunk 14]
```diff
@@ -10,68 +10,79 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
 - #### Step 1: Configure the DNS record of your domain, which is denoted with `CUSTOM_DOMAIN`.
 Add a `CNAME` entry for your domain pointing to `CUSTOM_DOMAIN.icp1.io` such that all the traffic destined to your domain is redirected to the boundary nodes;
 Add a `TXT` entry containing the canister ID to the `_canister-id`-subdomain of your domain (e.g., `_canister-id.CUSTOM_DOMAIN`);
 Add a `CNAME` entry for the `_acme-challenge`-subdomain (e.g., `_acme-challenge.CUSTOM_DOMAIN`) pointing to `_acme-challenge.CUSTOM_DOMAIN.icp2.io` in order for the boundary nodes to acquire the certificate.
 
-:::info
-In many cases, it is not possible to set a `CNAME` record for the top of a domain, the Apex record. In this case, DNS providers support so-called `CNAME` flattening. To this end, these DNS providers offer flattened record types, such as `ANAME` or `ALIAS` records, which can be used instead of the `CNAME` to `CUSTOM_DOMAIN.icp1.io`. The [following guide](dns-setup.mdx) provides detailed instructions for three popular registrars.
+:::caution
+
+These records must be configured before continuing the next steps.
```
Comment (r-birkner) [inline docs/developer-docs/web-apps/custom-domains/using-custom-domains.mdx:44]:
```suggestion
These records must be configured before continuing with the next steps.
```

**Comment by r-birkner** [comment]
Left some comments, but most of them are more like my suggestion. However, I think the domain names need to match the screenshots. So, those comments should be addressed.

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 5 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:89]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 7 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:117]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 8 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:122]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 9 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:145]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 10 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:150]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 11 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:154]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 12 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:155]: Updated screenshots

File: docs/developer-docs/web-apps/custom-domains/dns-setup.mdx (hunk 13 above)
Comment (jessiemongeon1) [reply docs/developer-docs/web-apps/custom-domains/dns-setup.mdx:160]: Updated screenshots

**Pull request merged**